fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/model/network/method/message/data/federation/FederationResultTest.kt

```diff
@@ -0,0 +1,137 @@
+        Assert.assertEquals(MG_CHALLENGE, RESULT_FAILURE.challenge)
+    }
+
+    @Test
+    fun resultObjectTest() {
+        Assert.assertEquals(Objects.FEDERATION.`object`, RESULT_SUCCESS.`object`)
+        Assert.assertEquals(Objects.FEDERATION.`object`, RESULT_FAILURE.`object`)
+    }
+
+    @Test
+    fun resultActionTest() {
+        Assert.assertEquals(Action.RESULT.action, RESULT_SUCCESS.action)
+        Assert.assertEquals(Action.RESULT.action, RESULT_FAILURE.action)
+    }
+
+    @Test
+    fun resultEqualsTest() {
+        val result2 = FederationResult(SUCCESS, publicKey = PK.encoded, challenge = MG_CHALLENGE)
+        val result3 = FederationResult(FAILURE, reason = REASON, challenge = MG_CHALLENGE)
+        val result4 = FederationResult(FAILURE, reason = "reason2", challenge = MG_CHALLENGE)
+        Assert.assertEquals(RESULT_SUCCESS, result2)
+        Assert.assertEquals(RESULT_SUCCESS, RESULT_SUCCESS)
+        Assert.assertEquals(RESULT_SUCCESS.hashCode().toLong(), result2.hashCode().toLong())
+        Assert.assertEquals(RESULT_FAILURE, result3)
+        Assert.assertEquals(RESULT_FAILURE, RESULT_FAILURE)
+        Assert.assertEquals(RESULT_FAILURE.hashCode().toLong(), result3.hashCode().toLong())
+
+        Assert.assertNotEquals(RESULT_SUCCESS, RESULT_FAILURE)
+        Assert.assertNotEquals(RESULT_SUCCESS, result3)
+        Assert.assertNotEquals(RESULT_SUCCESS, result4)
+        Assert.assertNotEquals(RESULT_FAILURE, result2)
+        Assert.assertNotEquals(RESULT_FAILURE, result4)
+        Assert.assertNotEquals(RESULT_SUCCESS, null)
+        Assert.assertNotEquals(RESULT_FAILURE, null)
+    }
+
+    @Test
+    fun resultToStringTest() {
+        Assert.assertEquals(
+                "FederationResult{status='$SUCCESS', public_key='${PK.encoded}', challenge='$MG_CHALLENGE'}",
+                RESULT_SUCCESS.toString()
+        )
+        Assert.assertEquals(
+                "FederationResult{status='$FAILURE', reason='$REASON', challenge='$MG_CHALLENGE'}",
+                RESULT_FAILURE.toString()
+        )
+    }
+
+    @Test
+    fun invalidMessageTypeTest() {
+        val exception = assertThrows(IllegalArgumentException::class.java) {
+            FederationResult("invalid", challenge = MG_CHALLENGE)
+        }
+        assert(exception.message == "Status must be either 'failure' or 'success'.")
+    }
+
+    @Test
+    fun invalidSuccessTest1() {
+        val exception = assertThrows(IllegalArgumentException::class.java) {
+            FederationResult(SUCCESS, challenge = MG_CHALLENGE)
+        }
+        assert(exception.message == "Public key must be provided for success status.")
+    }
+
+    @Test
+    fun invalidSuccessTest2() {
+        val exception = assertThrows(IllegalArgumentException::class.java) {
+            FederationResult(SUCCESS, publicKey = PK.encoded, reason = "reason", challenge = MG_CHALLENGE)
+        }
+        assert(exception.message == "Reason must be null for success status.")
```

Comment:
maybe we want to add those strings in Constants to avoid failing if the message comes to change